src/components/Editor.js

```diff
@@ -63,20 +63,19 @@ function resizeBreakpointGutter(editor) {
   breakpoints.style.width = `${lineNumbers.clientWidth}px`;
 }
 
-function traverseResults(e, ctx, dir) {
+function traverseResults(
+  e, ctx, query, dir, caseSensitive, wholeWord, regexMatch) {
   e.stopPropagation() || e.preventDefault();
-  const query = ctx.cm.getSelection();
   if (dir == "prev") {
-    findPrev(ctx, query);
+    findPrev(ctx, query, caseSensitive, wholeWord, regexMatch);
   } else if (dir == "next") {
-    findNext(ctx, query);
+    findNext(ctx, query, caseSensitive, wholeWord, regexMatch);
   }
 }
 
-function onCursorActivity(ctx) {
+function onCursorActivity(ctx, query, caseSensitive, wholeWord, regexMatch) {
   if (ctx.cm.somethingSelected()) {
-    const query = ctx.cm.getSelection();
-    find(ctx, query, true);
+    find(ctx, query, true, caseSensitive, wholeWord, regexMatch);
```

Comment:
`keepSelection` logic was causing some weirdness where it would reset the query to a blank string. I'll see if I can fix that and add that back in.